nautobot/dcim/models/device_component_templates.py

```diff
@@ -84,7 +85,20 @@ def instantiate_model(self, model, device, **kwargs):
         """
         Helper method to self.instantiate().
         """
-        return model(device=device, name=self.name, label=self.label, description=self.description, **kwargs)
+        content_type = ContentType.objects.get_for_model(model)
+        fields = CustomField.objects.filter(content_types=content_type)
+        for field in fields:
+            self._custom_field_data.setdefault(field.name, field.default)
```

Comment:
I'm not sure that storing the instance's custom field defaults onto `self` makes sense here. In particular a ComponentTemplateModel may theoretically have its own custom fields that are not actually applicable to the related ComponentModel - probably better to just construct a dict specifically for the `model` that can be passed through to the constructor and then discarded.